src/Zynga/Framework/ShardedDatabase/V3/Config/Base.hh

```diff
@@ -139,10 +140,14 @@ abstract class Base<TType as TypeInterface> extends FactoryBaseConfig
     throw new InvalidShardIdException('offset='.$offset);
   }
 
-  public function getShardId(TType $intShardType): int {
-    if ($intShardType instanceof UInt64Box) {
-      $shardCount = $this->getServerCount();
-      $shardId = $intShardType->get() % $shardCount;
+  public function getShardId(TType $shardType): int {
+    $shardCount = $this->getServerCount();
+    if ($shardType instanceof UInt64Box) {
+      $shardId = $shardType->get() % $shardCount;
+      return $shardId;
+    } else if($shardType instanceof StringBox) {
+      $crcString = crc32($shardType->get());
```

Comment:
timely change. We were doing this in our PgModel class for an emailHash. crcString seems misnamed (or is it just me)? crcValue, stringCRC, ...?

Reply:
ohh this is for getting shard by email id, lol I was thinking zid as a StringBox and both cases may return diff shard numbers.

Reply:
as long as they are always the same case when evaluating, maybe make sure they are lowercased here before running the emails through crc32.